### PR TITLE
Replace `default` with `function` to refer function's default worker

### LINKED ISSUE #28625: Use `function` to refer to function's default worker
**Description:**
$subject. 

Lang issue to track: ballerina-platform/ballerina-spec#707

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.
With this change, "default" will no longer be a reserved keyword in the language.

Fixes #28625

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [x] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -11652,8 +11637,8 @@ private STNode parseReceiveFieldEnd() {
      */
     private STNode parseReceiveField() {
         switch (peek().kind) {
-            case DEFAULT_KEYWORD:
-                STNode defaultKeyword = parseDefaultKeyword();
+            case FUNCTION_KEYWORD:
+                STNode defaultKeyword = consume();
                 return STNodeFactory.createSimpleNameReferenceNode(defaultKeyword);
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:11642]:
```suggestion
            case FUNCTION_KEYWORD:
                STNode functiontKeyword = consume();
                return STNodeFactory.createSimpleNameReferenceNode(functiontKeyword);
```

File: tests/jballerina-unit-test/src/test/resources/test-src/workers/invalid-receive-before-workers.bal [hunk 2]
```diff
@@ -2,6 +2,6 @@ function invalidWorkReceiveBeforeWorker() {
   int m1 = <- w1;
   worker w1 {
     int i = 1;
-    i -> default;
+    i -> function;
   }
 }
```
Comment (rdulmina) [inline tests/jballerina-unit-test/src/test/resources/test-src/workers/invalid-receive-before-workers.bal:7]: Let's add missing new lines of these files with this PR. Please fix in other files as well

**Review by rdulmina — COMMENTED** [review]
Parser changes LGTM

**Review by MaryamZi — APPROVED** [review]

**Comment by MaryamZi** [comment]
@lochana-chathura, I think we can also close https://github.com/ballerina-platform/ballerina-lang/issues/23476 now since it is no longer valid.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangInvokableNode.java [hunk 3]
```diff
@@ -42,7 +42,7 @@
  */
 public abstract class BLangInvokableNode extends BLangNode implements InvokableNode {
 
-    public static final String DEFAULT_WORKER_NAME = "default";
+    public static final String DEFAULT_WORKER_NAME = "function";
```
Comment (KRVPerera) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangInvokableNode.java:45]: Unfortunately, there are some more `DEFAULT_WORKER_NAME` constants with "default". Shall we remove them if they are not being used? 
`BLangConstants.java` is one example.

Comment (KRVPerera) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangInvokableNode.java:45]: Got merged while I was reviewing :). Shall we create an issue for this?

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangInvokableNode.java:45]: @KRVPerera will address this when addressing https://github.com/ballerina-platform/ballerina-lang/issues/23476#issuecomment-783488634. 
Added comment there. https://github.com/ballerina-platform/ballerina-lang/issues/23476#issuecomment-783856672

**Pull request merged**